src/librustc_trans/trans/adt.rs

```diff
@@ -438,23 +469,33 @@ struct Case<'tcx> {
 /// This represents the (GEP) indices to follow to get to the discriminant field
 pub type DiscrField = Vec<usize>;
 
-fn find_discr_field_candidate<'tcx>(tcx: &ty::ctxt<'tcx>,
+fn find_discr_field_candidate<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
                                     ty: Ty<'tcx>,
-                                    mut path: DiscrField) -> Option<DiscrField> {
+                                    mut path: DiscrField) -> Option<(DiscrField, ValueRef)> {
+    let ref tcx = cx.tcx();
     match ty.sty {
-        // Fat &T/&mut T/Box<T> i.e. T is [T], str, or Trait
+        // Fat &T/&mut T/Box<T> i.e. T is [T], str, or Trait: null is forbidden
         ty::TyRef(_, ty::TypeAndMut { ty, .. }) | ty::TyBox(ty) if !type_is_sized(tcx, ty) => {
             path.push(FAT_PTR_ADDR);
-            Some(path)
+            Some((path, C_null(type_of::sizing_type_of(cx, ty))))
```

Comment:
Mmmh... Ok. Where in the code can I convince myself that we only use a single byte as discriminant?